cnxarchive/views.py

```diff
@@ -286,6 +293,67 @@ def _get_content_json(environ=None, ident_hash=None, reqtype=None):
 
     return result
 
+def html_date(datetime):
+    """
+    Returns the HTTP-date format of python's datetime time as per:
+    http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html
+    """
+    weekdays = {0:'Mon', 1:'Tue', 2:'Wed', 3:'Thu', 4:'Fri', 5:'Sat', 6:'Sun'}
+    months = {'01': 'Jan', '02': 'Feb', '03': 'Mar', '04': 'Apr', '05': 'May', '06': 'Jun', '07': 'Jul', '08': 'Aug', '09': 'Sept', '10': 'Oct', '11': 'Nov', '12': 'Dec'}
+    datetime_str = datetime.__str__()
+
+    day = datetime_str.split()[0].split('-')[2]
+    month = datetime_str.split()[0].split('-')[1]
+    year = datetime_str.split()[0].split('-')[0]
+    time = datetime_str.split()[1][:8]
+
+    return weekdays[datetime.weekday()] + ', ' + day + ' ' + months[month] + ' ' + year + ' ' + time + ' GMT'
```

Comment:
There are better ways to format a string. Have you looked at https://docs.python.org/3/library/datetime.html#strftime-strptime-behavior ?